python/tk_multi_publish/publish.py

```diff
@@ -361,23 +389,29 @@ def _scan_scene(self):
                 
         return items
         
-    def _do_pre_publish(self, primary_task, secondary_tasks, progress_cb):
+    def _do_pre_publish(self, primary_task, secondary_tasks, progress_cb, user_data):
         """
         Do pre-publish pass on tasks using the pre-publish hook
         """
         
         # do pre-publish of primary task:
-        primary_task.pre_publish_errors = self._app.execute_hook("hook_primary_pre_publish",  
-                                                                task=primary_task.as_dictionary(), 
-                                                                work_template = self._work_template,
-                                                                progress_cb=progress_cb)
+        primary_task.pre_publish_errors = self._app.execute_hook(
+            "hook_primary_pre_publish",  
+            task=primary_task.as_dictionary(), 
+            work_template=self._work_template,
+            progress_cb=progress_cb,
+            user_data=user_data,
```

Comment:
You're adding a new argument passed to the hooks. If someone installs this new version of the app and they already have custom hooks, will this raise a `too many argument's` type exception if their hooks aren't expecting/accepting additonal keyword arguments? 

It looks like the default hooks all have `**kwargs` for additional arguments, but it's at least worth testing to see what happens if they have a hook that doesn't capture them. It would also be good to add to the docs how to update existing hooks to account for the change.